twitter.py

```diff
@@ -1,59 +1,62 @@
 import base64
+import collections
 import json
 
-import requests
 from requests_oauthlib import OAuth1
 
 from environment_variables import get_env
+from request import request
+
+endpoint = collections.namedtuple('twitter_method', ["url", "type"])
 
 
 class Twitter:
-    method = {"twitter_upload_pic": "https://upload.twitter.com/1.1/media/upload.json",
-              "twitter_create_post": "https://api.twitter.com/1.1/statuses/update.json",
-              "twitter_get_users_posts": "https://api.twitter.com/1.1/statuses/user_timeline.json",
-              "delete_tweet_by_id": "https://api.twitter.com/1.1/statuses/destroy/{}.json"}
-
-    method_type = {"twitter_upload_pic": "POST",
-                   "twitter_create_post": "POST",
-                   "twitter_get_users_posts": "GET",
-                   "delete_tweet_by_id": "POST"}
-
-    auth_post = OAuth1(get_env("TWITTER_CONSUMER_KEY"),
-                       client_secret=get_env("TWITTER_CONSUMER_SECRET"),
-                       resource_owner_key=get_env("TWITTER_ACCESS_KEY"),
-                       resource_owner_secret=get_env("TWITTER_ACCESS_SECRET"), signature_type="body")
+    twitter_upload_pic = endpoint("https://upload.twitter.com/1.1/media/upload.json", "POST")
+    twitter_create_post = endpoint("https://api.twitter.com/1.1/statuses/update.json", "POST")
+    twitter_get_users_posts = endpoint("https://api.twitter.com/1.1/statuses/user_timeline.json", "GET")
+    twitter_delete_tweet_by_id = endpoint("https://api.twitter.com/1.1/statuses/destroy/{}.json", "POST")
+
+    auth = OAuth1(get_env("TWITTER_CONSUMER_KEY"),
+                  client_secret=get_env("TWITTER_CONSUMER_SECRET"),
+                  resource_owner_key=get_env("TWITTER_ACCESS_KEY"),
+                  resource_owner_secret=get_env("TWITTER_ACCESS_SECRET"), signature_type="body")
     auth_get = OAuth1(get_env("TWITTER_CONSUMER_KEY"),
                       client_secret=get_env("TWITTER_CONSUMER_SECRET"),
                       resource_owner_key=get_env("TWITTER_ACCESS_KEY"),
                       resource_owner_secret=get_env("TWITTER_ACCESS_SECRET"))
 
-    def request(self, method_name, **kwargs):
-        result = None
-        if self.method_type[method_name] == "POST":
-            result = requests.post(self.method[method_name].format(kwargs.get("tweet_id", "")),
-                                   data=kwargs,
-                                   auth=self.auth_post)
-            # format required for delete_tweet_by_id method
-        elif self.method_type[method_name] == "GET":
-            result = requests.get(self.method[method_name], params=kwargs, auth=self.auth_get)
-        return result
-
     def upload_photo(self, name, data):
         data = base64.b64encode(data)
-        upload_photo_result = self.request("twitter_upload_pic", name=name, media_data=data)
+        payload = {"name": name, "media_data": data}
```

Comment:
No need to create a separate var here, just pass dict directly to "payload" in request()

Reply:
also fixed in flickr